fog/ledger/server/src/server.rs

```diff
@@ -213,3 +230,20 @@ impl<E: LedgerEnclaveProxy, R: RaClient + Send + Sync + 'static> Drop for Ledger
         self.stop();
     }
 }
+
+/// State that we want to expose from the db poll thread
+#[derive(Debug, Default)]
+pub struct DbPollSharedState {
+    /// The highest block count for which we can guarantee we have loaded all
+    /// available data.
+    pub highest_processed_block_count: u64,
+
+    /// A block signature timestamp for the highest processed block
+    pub highest_processed_block_signature_timestamp: u64,
+
+    /// The last block count for which we were able to load data.
+    pub last_known_block_count: u64,
```

Comment:
for instance, if the server has just started up, `last_known_block_count` is the same as `ledger_db.num_blocks()` but `highest_processed_block_count` would be zero because this fetcher thread hasn't put anything in enclave yet